client/src/modules/cost_center/allocation_bases/allocation_bases.js

```diff
@@ -0,0 +1,107 @@
+angular.module('bhima.controllers')
+  .controller('CostCenterAllocationBasesController', CostCenterAllocationBasesController);
+
+CostCenterAllocationBasesController.$inject = [
+  'CostCenterService',
+  'uiGridConstants',
+  'NotifyService',
+];
+
+function CostCenterAllocationBasesController(
+  CostCenters,
+  uiGridConstants,
+  Notify,
+) {
+  const vm = this;
+
+  // global variables
+  vm.gridApi = {};
+  vm.filterEnabled = false;
+
+  // options for the UI grid
+  vm.gridOptions = {
+    appScopeProvider  : vm,
+    enableColumnMenus : false,
+    showColumnFooter  : true,
+    fastWatch         : true,
+    flatEntityAccess  : true,
+    enableSorting     : false,
+    onRegisterApi     : onRegisterApiFn,
+  };
+
+  function onRegisterApiFn(gridApi) {
+    vm.gridApi = gridApi;
+  }
+
+  function load() {
+    CostCenters.getAllocationKeys()
+      .then(rows => {
+        if (!rows || (rows && !rows.costCenterIndexes.length)) { return; }
+
+        const columnDefs = [{
+          field : 'label',
+          displayName : 'COST_CENTER.TITLE',
+          headerCellFilter : 'translate',
+          cellFilter : 'translate',
+          aggregationHideLabel : true,
+          aggregationType : uiGridConstants.aggregationTypes.count,
+        }];
+
+        rows.costCenterIndexes.forEach((item, idx) => {
+          if (idx >= 0) {
```

Comment:
If I am not mistaken, the index in forEach Will always be positive.  Why do you need this check?